logic/transactions/pool.js

```diff
@@ -0,0 +1,922 @@
+		}
+		// total payments
+		paymentTransactions = self.getAll('sender_id', { id: sender.address });
+		if (paymentTransactions.ready.length > 0) {
+			paymentTransactions.ready.forEach(function (paymentTransaction) {
+				if (paymentTransaction.amount) {
+					poolBalance = poolBalance.minus(paymentTransaction.amount.toString());
+				}
+				poolBalance = poolBalance.minus(paymentTransaction.fee.toString());
+			});
+		}
+		
+		// total receipts
+		receiptTransactions = self.getAll('recipient_id', { id: sender.address });
+		if (receiptTransactions.ready.length > 0) {
+			receiptTransactions.ready.forEach(function (receiptTransaction) {
+				if (receiptTransaction.type === transactionTypes.SEND) {
+					poolBalance = poolBalance.plus(receiptTransaction.amount.toString());
+				}
+			});
+		}
+
+		// total balance
+		var balance = new bignum(account.balance.toString());
+		balance = balance.plus(poolBalance);
+
+		// Check confirmed sender balance
+		var amount = new bignum(transaction.amount.toString()).plus(transaction.fee.toString());
+		var exceeded = balance.lessThan(amount);
+
+		if (exceeded) {
+			return setImmediate(cb, [
+				'Account does not have enough LSK:', sender.address,
+				'balance:', balance.div(Math.pow(10,8))
+			].join(' '));
+		}
+
+		return setImmediate(cb, null, [
+			'balance:', balance.div(Math.pow(10,8))
+		].join(' '));
+	});
+};
+
+/**
+ * Validates a transaction and adds it to the verified pool list.
+ * @implements {__private.addToVerified}
+ * @param {transaction} transaction
+ * @param {boolean} broadcast
+ * @param {function} cb - Callback function.
+ * @return {setImmediateCallback} error | cb
+ */
+TransactionPool.prototype.addFromPublic = function (transaction, broadcast, cb) {
+	__private.addToVerified(transaction, broadcast, cb);
+};
+
+/**
+ * Adds transactions to unverified pool list.
+ * @implements {__private.addToUnverified}
+ * @param {transaction} transactions
+ * @param {boolean} broadcast
+ * @param {function} cb - Callback function.
+ * @return {setImmediateCallback} error | cb
+ */
+TransactionPool.prototype.addFromPeer = function (transactions, broadcast, cb) {
+	if (!Array.isArray(transactions)) {
+		transactions = [transactions];
+	}
+	async.eachSeries(transactions, function (transaction, cb) {
+		__private.addToUnverified(transaction, broadcast, cb);
+	}, function (err) {
+		return setImmediate(cb, err);
+	});
+};
+
+/**
+ * Adds transactions to verified.ready pool list.
+ * @implements {__private.addToPoolList}
+ * @implements {delete}
+ * @param {transaction} transactions
+ * @param {function} cb - Callback function.
+ * @return {setImmediateCallback} error | cb
+ */
+TransactionPool.prototype.addReady = function (transactions, cb) {
+	if (!Array.isArray(transactions)) {
+		transactions = [transactions];
+	}
+	var resetReceivedAt = new Date();
+	async.eachSeries(transactions, function (transaction, eachSeriesCb) {
+		self.delete(transaction);
+		transaction.receivedAt = resetReceivedAt;
```

Comment:
Why overwrite that here?

Reply:
The field `receivedAt` is not part of these transactions, because they come from a rollback block, `resetReceivedAt` is to show that this is not the first time that transaction access to the pool. 
Maybe we won't have to use this function